src/main/java/com/minecraftabnormals/abnormals_core/core/api/model/FullbrightModelLoader.java

```diff
@@ -0,0 +1,259 @@
+                    throw new JsonParseException("Invalid rotation " + angle + " found, only -45/-22.5/0/22.5/45 allowed");
+                }
+
+                return angle;
+            }
+
+            private Direction.Axis parseAxis(JsonObject object) {
+                String axisName = JSONUtils.getString(object, "axis");
+                Direction.Axis axis = Direction.Axis.byName(axisName.toLowerCase(Locale.ROOT));
+                if (axis == null) {
+                    throw new JsonParseException("Invalid rotation axis: " + axisName);
+                }
+
+                return axis;
+            }
+
+            private Map<Direction, BlockPartFace> parseFacesCheck(JsonObject object) {
+                Map<Direction, BlockPartFace> faces = this.parseFaces(object);
+                if (faces.isEmpty()) {
+                    throw new JsonParseException("Expected between 1 and 6 unique faces, got 0");
+                }
+
+                return faces;
+            }
+
+            private Map<Direction, BlockPartFace> parseFaces(JsonObject object) {
+                Map<Direction, BlockPartFace> faces = Maps.newEnumMap(Direction.class);
+                JsonObject facesObject = JSONUtils.getJsonObject(object, "faces");
+                for (Map.Entry<String, JsonElement> entry : facesObject.entrySet()) {
+                    Direction direction = this.parseEnumFacing(entry.getKey());
+                    faces.put(direction, GSON.fromJson(entry.getValue(), FullbrightBlockPartFace.class));
+                }
+
+                return faces;
+            }
+
+            private Direction parseEnumFacing(String name) {
+                Direction direction = Direction.byName(name);
+                if (direction == null) {
+                    throw new JsonParseException("Unknown facing: " + name);
+                }
+
+                return direction;
+            }
+
+            private Vector3f validateVectorBounds(JsonObject object, String name) {
+                Vector3f vector = this.deserializeVec3f(object, name);
+                if (!(vector.getX() < -16.0F) && !(vector.getY() < -16.0F) && !(vector.getZ() < -16.0F) && !(vector.getX() > 32.0F) && !(vector.getY() > 32.0F) && !(vector.getZ() > 32.0F)) {
+                    return vector;
+                }
+
+                throw new JsonParseException("'" + name + "' specifier exceeds the allowed boundaries: " + vector);
+            }
+
+            private Vector3f deserializeVec3f(JsonObject object, String name) {
+                JsonArray vectorJson = JSONUtils.getJsonArray(object, name);
+                if (vectorJson.size() != 3) {
+                    throw new JsonParseException("Expected 3 " + name + " values, found: " + vectorJson.size());
+                }
+
+                float[] vector = new float[3];
+
+                for (int i = 0; i < vector.length; ++i) {
```

Comment:
The size of the array is constant.
```suggestion
                for (int i = 0; i < 3; ++i) {
```